src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/FinalizeStep/FinalizeStep.tsx

```diff
@@ -1,10 +1,126 @@
+import { ActionTypes } from '~redux';
+
+import { formatText } from '~utils/intl';
+import Button, { PendingButton } from '~v5/shared/Button';
+import { ActionForm } from '~shared/Fields';
+import CardWithStatusText from '~v5/shared/CardWithStatusText';
+import DescriptionList from '../VotingStep/partials/DescriptionList';
+import { useClaimConfig, useFinalizeStep } from './hooks';
+import { FinalizeStepProps } from './types';
+// import TeamBadge from '~v5/common/Pills/TeamBadge';
+import { useAppContext, useColonyContext } from '~hooks';
+import Icon from '~shared/Icon';
+
+import { ActionButton } from '~shared/Button';
 
 const displayName =
   'v5.common.ActionSidebar.partials.motions.MotionSimplePayment.steps.FinalizeStep';
 
-const FinalizeStep: FC = () => {
-  return <div>FinalizeStep</div>;
+const FinalizeStep: FC<FinalizeStepProps> = ({
+  actionData,
+  startPollingAction,
+  stopPollingAction,
+  refetchAction,
+}) => {
+  const { user } = useAppContext();
+  const [isPolling, setIsPolling] = useState(false);
+  const { refetchColony } = useColonyContext();
+  const { isFinalizable, transform } = useFinalizeStep(actionData);
+  const {
+    items,
+    isClaimed,
+    buttonTextId,
+    remainingStakesNumber,
+    handleClaimSuccess,
+    claimPayload,
+    canClaimStakes,
+  } = useClaimConfig(actionData, startPollingAction, refetchAction);
+
+  const handleSuccess = () => {
+    startPollingAction(1000);
+    setIsPolling(true);
+  };
+
+  /* Stop polling when mounted / dismounted */
+  useEffect(() => {
+    stopPollingAction();
+    return stopPollingAction;
+  }, [stopPollingAction]);
+
+  /* Update colony object when motion gets finalized. */
+  if (actionData.motionData.isFinalized) {
+    refetchColony();
+  }
+
+  if (isClaimed) {
+    stopPollingAction();
+  }
+
+  return (
+    <CardWithStatusText
+      statusTextSectionProps={{
+        status: 'info',
+        children: formatText({ id: 'motion.finalizeStep.statusText' }),
+        textClassName: 'text-4',
+        iconAlignment: 'top',
+        content: (
+          <span className="flex items-center text-4 mt-2">
+            <span className="flex text-blue-400 mr-2">
+              <Icon name="arrows-clockwise" appearance={{ size: 'tiny' }} />
+            </span>
+            {formatText(
+              { id: 'motion.finalizeStep.transactions.remaining' },
+              { transactions: remainingStakesNumber },
+            )}
+          </span>
+        ),
+      }}
+      sections={[
+        {
+          key: '1',
+          content: (
+            <ActionForm
+              actionType={ActionTypes.MOTION_FINALIZE}
+              transform={transform}
+              onSuccess={handleSuccess}
+            >
+              <div className="mb-2">
+                <h4 className="text-1 mb-3 flex justify-between items-center">
+                  {formatText({ id: 'motion.finalizeStep.title' })}
+                  {/* <TeamBadge
```

Comment:
I've added motion partial `MotionBadge`, please check if you can use it here

Reply:
I'll  add comment to use it here later. Currently the design should be updated and then I'll back to this motion step.